### PR TITLE
[16.0][MIG] resource_booking: Migration to 16.0

### PR DESCRIPTION
**Pull request by victoralmau**
Superseed https://github.com/OCA/calendar/pull/101

Migration to 16.0

Changes done:
- [x] Squash administrative commits.
- [x] Purge translations.
- [x] Fix tests and/or remove warnings.

Please @pedrobaeza and @carolinafernandez-tecnativa can you review it?

@Tecnativa TT45643

### CODE REVIEW

File: resource_booking/tests/test_portal.py [hunk 1]
```diff
@@ -7,33 +7,29 @@
 from freezegun import freeze_time
 from lxml.html import fromstring
 
+import odoo
```
Comment (pedrobaeza) [inline resource_booking/tests/test_portal.py]: Avoid this general import importing and using just `tagged`

File: resource_booking/tests/test_backend.py [hunk 2]
```diff
@@ -645,22 +645,22 @@ def test_videocall_location(self):
         with Form(rb) as rb_f:
             rb_f.start = "2021-03-01 08:00:00"
         self.assertEqual(rb.state, "scheduled")
-        self.assertEqual(rb.videocall_location, "Videocall Office 1")
-        self.assertEqual(rb.meeting_id.videocall_location, "Videocall Office 1")
+        self.assertTrue("Videocall Office 1" in rb.videocall_location)
```
Comment (pedrobaeza) [inline resource_booking/tests/test_backend.py]: Use `assertIn` instead (in all occurrences). Anyway, I don't get why you are changing this.

File: resource_booking/tests/test_backend.py [hunk 3]
```diff
@@ -268,9 +268,9 @@ def test_booking_from_calendar_view(self):
         self.assertEqual(self.rbt.duration, 0.5)
         # Change it to 45 minutes
         self.rbt.duration = 0.75
-        # Bookings smart button configures calendar with slots of 45 minutes
+        # Bookings smart button configures calendar with slots from slot duration field
         button_context = self.rbt.action_open_bookings()["context"]
-        self.assertEqual(button_context["calendar_slot_duration"], "00:45")
+        self.assertEqual(button_context["calendar_slot_duration"], "00:30:00")
```
Comment (pedrobaeza) [inline resource_booking/tests/test_backend.py:273]: Why this change in the test?

File: resource_booking/migrations/16.0.1.0.0/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from openupgradelib import openupgrade
+
+
+def _update_attendance_hour_to(env):
+    # 23:59 -> 24:00
+    lines = env["resource.calendar.attendance"].search(
```
Comment (pedrobaeza) [inline resource_booking/migrations/16.0.1.0.0/post-migration.py:10]: Why is this needed?

File: resource_booking/migrations/16.0.1.0.0/pre-migration.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from openupgradelib import openupgrade
+
+xmlids_spec = [
```
Comment (pedrobaeza) [inline resource_booking/migrations/16.0.1.0.0/pre-migration.py]: Why renaming this?

File: resource_booking/models/resource_booking.py [hunk 6]
```diff
@@ -137,6 +172,14 @@ class ResourceBooking(models.Model):
         tracking=True,
         help="Who will attend this booking?",
     )
+    partner_ids = fields.Many2many(
```
Comment (pedrobaeza) [inline resource_booking/models/resource_booking.py]: This one shouldn't be added in this commit.

File: resource_booking/tests/common.py [hunk 7]
```diff
@@ -42,7 +42,7 @@ def create_test_data(obj):
                 "name": "Fridays",
                 "dayofweek": "4",
                 "hour_from": 0,
-                "hour_to": 23.99,
+                "hour_to": 24,
```
Comment (pedrobaeza) [inline resource_booking/tests/common.py:45]: This should go in the same direction as the migration script, but I don't understand why it's needed.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I think you can merge both commits of Henrik, and put a better first line commit message on the other.

File: resource_booking/tests/test_backend.py (hunk 2 above)
Comment (victoralmau) [reply resource_booking/tests/test_backend.py]: I have left `assertEqual`, although it is now necessary to specify http or https to avoid auto-defining with the base url.

File: resource_booking/migrations/16.0.1.0.0/pre-migration.py (hunk 5 above)
Comment (victoralmau) [reply resource_booking/migrations/16.0.1.0.0/pre-migration.py]: Removed.

File: resource_booking/models/resource_booking.py (hunk 6 above)
Comment (victoralmau) [reply resource_booking/models/resource_booking.py]: The final commit I added removes this (I didn't want to modify the author's commits).

Comment (pedrobaeza) [reply resource_booking/models/resource_booking.py]: I think this is confusing in this case. Better to not change it. And this should be already existing in 15.0 previous commits.

File: resource_booking/tests/test_backend.py (hunk 3 above)
Comment (victoralmau) [reply resource_booking/tests/test_backend.py:273]: All this change is related to the `slot_duration` field that is added in `resource.booking.type`, it changes with respect to how it is set in v15: https://github.com/OCA/calendar/blob/15.0/resource_booking/models/resource_booking_type.py#L179 Is it necessary? I think not because there are now 2 fields to reflect the same thing.

File: resource_booking/models/resource_booking.py (hunk 6 above)
Comment (victoralmau) [reply resource_booking/models/resource_booking.py]: Ok, I will remove it from the corresponding commit

File: resource_booking/tests/test_backend.py (hunk 3 above)
Comment (pedrobaeza) [reply resource_booking/tests/test_backend.py:273]: Well, that slot is though for presenting the calendar in intervals of such duration, so it has a goal, but not sure how it was handled before and now.

Comment (victoralmau) [reply resource_booking/tests/test_backend.py:273]: In v15 this already exists (with the `duration` field), so I comment that I don't understand the need for this extra field for "the same thing".

Comment (victoralmau) [reply resource_booking/tests/test_backend.py:273]: Ok, I see that it is related to https://github.com/OCA/calendar/pull/101#discussion_r1374273498 and is intentional.

File: resource_booking/tests/common.py (hunk 7 above)
Comment (victoralmau) [reply resource_booking/tests/common.py:45]: This change checks https://github.com/OCA/calendar/pull/120#discussion_r1519952482

File: resource_booking/migrations/16.0.1.0.0/post-migration.py (hunk 4 above)
Comment (victoralmau) [reply resource_booking/migrations/16.0.1.0.0/post-migration.py:10]: I have added a comment in the method to try to explain why this is necessary.

**Comment by victoralmau** [comment]
Changes done.

**Review by jelenapoblet — APPROVED** [review]
Tested on runboat and on premise instance, working fine on my side.

**Comment by carolinafernandez-tecnativa** [comment]
I have tested, and after installing contacts module i got this error message

image

It is refering to this field: https://github.com/OCA/calendar/pull/120/commits/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R10

https://github.com/OCA/calendar/pull/120/commits/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R18

On this PR, there was a change to allow multiple partners in resource booking.
https://github.com/OCA/calendar/pull/110

Thanks!

**Comment by victoralmau** [comment]
> I have tested, and after installing contacts module i got this error message
> image
> 
> It is refering to this field: [b31f313#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R10](https://github.com/OCA/calendar/commit/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R10)
> 
> [b31f313#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R18](https://github.com/OCA/calendar/commit/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R18)
> 
> On this PR, there was a change to allow multiple partners in resource booking. #110
> 
> Thanks!

FIxed.

**Comment by carolinafernandez-tecnativa** [comment]
> > I have tested, and after installing contacts module i got this error message
> > image
> > It is refering to this field: [b31f313#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R10](https://github.com/OCA/calendar/commit/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R10)
> > [b31f313#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R18](https://github.com/OCA/calendar/commit/b31f3136c0987800f7affeb8cbfcf15c89768730#diff-b082390b74a6351a8e33162e48a73b9ee0840757e7a7006440ca5eea629705c1R18)
> > On this PR, there was a change to allow multiple partners in resource booking. #110
> > Thanks!
> 
> FIxed.

Thanks, it is solved.

When im trying to book for this resource, i got this error message:

image


Check this configuration:

image

image

image

Maybe it there any conflict with timezone?

**Comment by victoralmau** [comment]
Checking on this, I see that there is no problem, although there is some confusion.

In runboat the Admin user has the Timezone America/Montevideo, therefore, when you want to set a start date for the booking, you are doing it with that date and time in America/Montevideo, but if we "_convert_" the date and time it would not be available in the corresponding work schedule.

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Functional Review, LGTM :) Thanks!

File: resource_booking/models/res_partner.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    resource_booking_count = fields.Integer(
+        compute="_compute_resource_booking_count", string="Resource booking count"
+    )
+    resource_booking_ids = fields.One2many(
+        "resource.booking", "partner_id", string="Bookings"
```
Comment (pedrobaeza) [inline resource_booking/models/res_partner.py]: This one is incorrect, as it only reflects the first attendee. It should be a computed one.

**Review by pedrobaeza — COMMENTED** [review]
The commit message `[IMP] resource_booking: add freezegun as a python external dependency` is incorrect, as it's the first one of a lot of changes. You should put a general commit message and then the summary. The same for the other one.

File: resource_booking/models/res_partner.py (hunk 8 above)
Comment (victoralmau) [reply resource_booking/models/res_partner.py]: Change to many2many field + default_partner_ids.

**Comment by victoralmau** [comment]
> The commit message `[IMP] resource_booking: add freezegun as a python external dependency` is incorrect, as it's the first one of a lot of changes. You should put a general commit message and then the summary. The same for the other one.

Commit message changed (summary + changes).

**Review by norlinhenrik — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Comment by tarteo** [comment]
@victoralmau Why did you change the ecma version?

**Comment by pedrobaeza** [comment]
Restored in #165
